staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/types.go

```diff
@@ -67,5 +69,20 @@ type Key struct {
 	Secret string `json:"secret"`
 }
 
+// KMSConfig contains the configuration for the cloudkms service provider to be used for
+// implementing the KMS based encryption transformer, as well as configuration for the
+// KMS transformer.
+type KMSConfig struct {
+	// kind is the name of the cloudkms Service provider to be used.
+	Kind string `json:"kind,omitempty"`
+	// apiVersion is the API version this block has to be parsed as.
+	APIVersion string `json:"apiVersion"`
+	// cacheSize is the maximum number of secrets which are cached in memory. The default value is 1000.
+	// +optional
+	CacheSize int `json:"cacheSize,omitempty"`
+	// config is provider specific configuration and is different for each cloudkms service provider.
+	Config map[string]interface{} `json:"config,omitempty"`
```

Comment:
If I understand your comment correctly, you are talking about moving the entire KMSConfig struct outside apiserver. If that is not so, please correct me.

So, the reason this code is here, is that the KMS transformer code is independent of the KMS provider, or the cloudprovider code. The cloudprovider specific code is a [new struct](https://github.com/sakshamsharma/kubernetes/blob/3c94f3ba6769afb7187799f15f75d6189b47289a/pkg/cloudprovider/providers/gce/gce_kms.go#L38), which is inside cloudproviders already. Adding a new KMS service is a matter of adding a new file in cloudproviders package, as shown by [this](https://github.com/sakshamsharma/kubernetes/blob/3c94f3ba6769afb7187799f15f75d6189b47289a/staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go#L152).

The KMS transformer would reside in k8s.io/apiserver/storage/value/encrypt/kms. It made sense (for me) to have the bare minimum configuration (which is, the above KMSConfig struct, containing the cache size and the name of the cloudkms) for that transformer in k8s.io/apiserver too.

If the above explanation is not acceptable, can you please give me a direction to work in, to make this PR acceptable?